Example/ViewController.swift

```diff
@@ -139,6 +137,40 @@ class ViewController: UIViewController, GalleryItemsDataSource, GalleryDisplaced
     }
 }
 
+extension ViewController: GalleryDisplacedViewsDataSource {
+
+    func provideDisplacementItem(atIndex index: Int) -> DisplaceableView? {
+
+        return index < imageViews.count ? imageViews[index] : nil
+    }
+}
+
+extension ViewController: GalleryItemsDataSource {
+
+    func itemCount() -> Int {
+
+        return imageViews.count
+    }
+
+    func provideGalleryItem(_ index: Int) -> GalleryItem {
+
+        let imageView = imageViews[index]
+        return items[imageView]!
+    }
+}
+
+extension ViewController: GalleryItemsDelegate {
+
+    func removeGalleryItem(at index: Int) {
+
+        print("remove item at \(index)")
+
+        let imageView = imageViews[index]
+        imageView.removeFromSuperview()
+        imageViews.remove(at: index)
```

Comment:
Since this is just test code, and not really how one would implement a view like this, I didn't see the need to keep the two structures in sync - the dictionary will always contain a superset of the data in the array, with no real downside.  If you prefer, I can come up with a cleaner and more representative example at the expense of additional code.